secedgar/cik_lookup.py

```diff
@@ -1,27 +1,47 @@
+import functools
 import warnings
 
-from secedgar.client.network_client import NetworkClient
-from secedgar.utils.cik_map import get_cik_map
-from secedgar.utils.exceptions import CIKError, EDGARQueryError
+import requests
 
+from secedgar.client import NetworkClient
+from secedgar.exceptions import CIKError, EDGARQueryError
 
-class _CIKValidator(object):
-    """Validates company tickers and/or company names based on CIK availability.
 
-    Used internally by the CIK class. Not intended for outside use.
+@functools.lru_cache()
```

Comment:
why this over just `functools.cache()`. Tho good to use built in caching. https://docs.python.org/3/library/functools.html

Reply:
`cache` was introduced in 3.9 and is essentially a wrapper around `lru_cache`.